spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/XDStreamParser.java

```diff
@@ -322,4 +333,98 @@ private ModuleType resolveModuleType(String moduleName, ModuleType... candidates
 		throw new NoSuchModuleException(moduleName, candidates);
 	}
 
+	/**
+	 * A wrapper around the canonical parser that supports multiline "document" parsing.
+	 *
+	 * <p>While XDStreamParser parses a single definition, this class is able to parse multiple, reporting
+	 * errors or accepting constructs like taps against definitions that appear earlier in the document (without
+	 * actually saving them in the main repository).</p>
+	 *
+	 * @author Eric Bottard
+	 */
+	public static class FloParser {
```

Comment:
I'd rather nothing we put into XD was called 'Flo' specific. I mean the content assist isn't FloContentAssist is it - it is a generally usable endpoint - shell and Flo both use it.